chaos_genius/alerts/anomaly_alerts.py

```diff
@@ -171,56 +251,33 @@ def _find_point(self, point, prev_data):
                 break
         return intended_point
 
-    def _save_nl_message_daily_freq(self, anomaly_data: List[dict], kpi: Kpi):
-        """Saves change message for every point, for a daily frequency KPI."""
-        time_diff = datetime.timedelta(days=1, hours=0, minutes=0)
-
-        # TODO: fix circular import
-        from chaos_genius.controllers.digest_controller import get_previous_data
-
-        prev_day_data = get_previous_data(kpi.id, self.anomaly_end_date, time_diff)
+    def _save_nl_message(self, anomaly_data: List[dict]):
+        """Constructs and saves change message for every point."""
+        kpi = get_kpi_obj(self.kpi_id)
+        if kpi is None:
+            for point in anomaly_data:
+                point["nl_message"] = "KPI does not exist"
 
-        prev_day_data = [anomaly_point.as_dict for anomaly_point in prev_day_data]
+            logger.info(f"KPI ID - {self.kpi_id} does not exist")
 
-        for point in prev_day_data:
-            if point.get("anomaly_type") != "overall":
-                point["series_type"] = convert_query_string_to_user_string(
-                    point["series_type"]
-                )
-            else:
-                point["series_type"] = OVERALL_KPI_SERIES_TYPE_REPR
+            return
 
-        for point in anomaly_data:
-            intended_point = self._find_point(point, prev_day_data)
-
-            if intended_point is None:
-                # previous point wasn't found
-                point["percentage_change"] = "–"
-            elif point["y"] == 0 and intended_point["y"] == point["y"]:
-                # previous data was same as current
-                point["percentage_change"] = "–"
-            elif intended_point["y"] == 0:
-                # previous point was 0
-                sign_ = "+" if point["y"] > 0 else "-"
-                point["percentage_change"] = sign_ + "inf"
-            else:
-                point["percentage_change"] = find_percentage_change(
-                    point["y"], intended_point["y"]
-                )
+        time_series_freq = kpi.anomaly_params.get("frequency")
+        if time_series_freq is None or time_series_freq not in ("D", "H"):
+            for point in anomaly_data:
+                point["nl_message"] = "Time series frequency not found or invalid"
 
-            point["nl_message"] = change_message_from_percent(
-                point["percentage_change"]
+            logger.info(
+                f"Time series frequency not found or invalid for Alert {self.alert_id}"
```

Comment:
This should be warning too and include both Alert and KPI ID:
```suggestion
            logger.info(
                f"(KPI: {self.kpi_id}, Alert: {self.alert_id}) Time series frequency not found or invalid."
```